scripts/ai_generation/converter.py

```diff
@@ -0,0 +1,461 @@
+import re
+from github import Github
+import time
+import yaml
+import subprocess
+import sys
+
+### USER INPUT NEEDED ###
+openai.api_type = "azure"
+openai.api_version = "2023-05-15"
+openai.api_base = os.getenv("OPENAI_API_BASE")  # Your Azure OpenAI resource's endpoint value
+openai.api_key = os.getenv("OPENAI_API_KEY")  # Your Azure OpenAI resource's API Key
+github_access_token = os.getenv("GitHub_Token") # Your GitHUB personal access token
+g = Github(login_or_token=github_access_token)
+relevant_azure_docs = ['https://learn.microsoft.com/en-us/azure/virtual-machines/linux/quick-create-cli', 'https://learn.microsoft.com/en-us/azure/virtual-machines/linux/tutorial-secure-web-server'] # Links to Azure docs that you want to convert to executable docs as a list
+### END OF USER INPUT ###
+
+script_path = os.path.realpath(__file__)
+
+# Get the directory of the current script
+script_dir = os.path.dirname(script_path)
+
+# Get the root directory of the project by going up three levels from the script directory
+root_dir = os.path.dirname(os.path.dirname(script_dir))
+
+# Define the absolute path to the AIDocs directory
+AI_Docs = os.path.join(root_dir, "scenarios", "ocd", "AIDocs")
+test_exec_docs = "None"
+
+if os.path.exists(AI_Docs):
+    test_exec_docs = input("Do you want to test current exec doc(s)? (one or all or no): ") 
+    while True:
+        if 'one' in test_exec_docs.lower():
+            valid_exec_doc = False
+            test_exec_docs = input("Enter the name of the exec doc you want to test: ")
+            test_exec_docs = [test_exec_docs]
+            for root, dirs, files in os.walk(AI_Docs):
+                for dir in dirs:
+                    if ''.join(test_exec_docs).lower() in dir.lower():
+                        valid_exec_doc = True
+            if not valid_exec_doc:
+                test_exec_docs = input("Invalid input. Do you want to test current exec doc(s)? (one or all or no): ")
+                break
+            else:
+                break
+        if 'all' in test_exec_docs.lower():
+            test_exec_docs = []
+            for root, dirs, files in os.walk(AI_Docs):
+                print((root, dirs, files))
+                for dir in dirs:
+                    if any('md' in file for file in files):
+                        test_exec_docs.append(dir)
+            break
+        if 'no' in test_exec_docs.lower():
+            test_exec_docs = ""
+            break
+        else:
+            test_exec_docs = input("Invalid input. Do you want to test current exec doc(s)? (one or all or no): ")
+
+allowed_commands_list = ['azurecli','bash', 'terraform', 'azure-cli-interactive', 'console', 'yaml']
+
+system_prompt_generate_exec_doc = """You are a Microsoft Azure expert. You write tutorial documentation designed to help new users use Azure seamlessly. You are concise in your descriptions and strictly follow markdown style of documentation. Adhere to the following format.
```

Comment:
Can all these prompts be placed into their own separate files and then sourced by the script at runtime? This will reduce the script size dramatically while retaining the same functionality.

Reply:
Want to keep them in one file so that people can use this outside of IE, only under 1 script